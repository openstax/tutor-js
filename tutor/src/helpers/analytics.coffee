_ = require 'underscore'

Routes = require '../routes'

{CurrentUserStore} = require '../flux/current-user'

DestinationHelper  = require '../helpers/routes-and-destinations'

# generate custom event data for routes
Events =
  viewTaskStep: ({courseId}) ->
    # track all work done on a course
    Analytics.sendEvent 'Course', 'Work', label: courseId

  viewStudentDashboard: ({courseId}) ->
    # compare activity between courses
    Analytics.sendEvent 'Student', 'Dashboard', label: courseId


# Translators convert a url like '/foo/bar/123/baz/1' into a simplified one like just '/foo/bar'
Translators =

  dashboard:            ({courseId}) -> "/student/choose-course/#{courseId}"
  viewPractice:         ({courseId}) -> "/student/practice/#{courseId}"
  viewPerformanceForecast:  ({courseId}) -> "/student/performance-forecast/#{courseId}"
  viewStudentDashboard: ({courseId}) -> "/student/dashboard/#{courseId}"

  calendarByDate:        ({courseId}) -> "/teacher/calendar/#{courseId}"
  viewTeacherPerformanceForecast:      ({courseId}) -> "/teacher/performance-forecast/#{courseId}"
  viewScores:            ({courseId}) -> "/teacher/student-scores/#{courseId}"
  courseSettings:        ({courseId}) -> "/teacher/roster/#{courseId}"
  editReading:           ({courseId}) -> "/teacher/assignment/edit/reading/#{courseId}"
  editHomework:          ({courseId}) -> "/teacher/assignment/edit/homework/#{courseId}"
  editExternal:          ({courseId}) -> "/teacher/assignment/edit/external/#{courseId}"
  createReading:         ({courseId}) -> "/teacher/assignment/create/reading/#{courseId}"
  createHomework:        ({courseId}) -> "/teacher/assignment/create/homework/#{courseId}"
  createExternal:        ({courseId}) -> "/teacher/assignment/create/external/#{courseId}"
  calendarViewPlanStats: ({courseId}) -> "/teacher/metrics/quick/#{courseId}"
  reviewTask:            ({courseId}) -> "/teacher/metrics/review/#{courseId}"

  viewReferenceBook:        ({courseId}) -> "/reference-view/#{courseId}"
  viewReferenceBookSection: ({courseId}) -> "/reference-view/#{courseId}"
  viewReferenceBookPage:    ({courseId}) -> "/reference-view/#{courseId}"

  # Task steps are viewed by both teacher and student with no difference in params
  viewTaskStep:         ({courseId}) ->
    role = CurrentUserStore.getCourseRole(courseId, true)
    "/#{role}/task-step/#{courseId}"


GA = undefined

Analytics =

  setTracker: (tracker) -> GA = tracker

  sendPageView: (url) ->
    GA?('send', 'pageview', url)

  onNavigation: (path) ->
    return unless GA

    route = Routes.pathToEntry(path)

    return @sendPageView("/not-found/#{change.path}") unless route

    path = Translators[route.entry.name]?( route.match.params ) or route.match.pathname

    # if we're also going to send custom events then we set the page
    if Events[route.entry.name]
      GA('set', 'page', path)
      Events[route.entry.name]( route.match.params )
      @sendPageView() # url's not needed since it was set before events
    else
      @sendPageView(route.match.pathname)

  sendEvent: (category, action, attrs) ->
    return unless GA
    GA('send', 'event', category, action, attrs.label, attrs.value)

module.exports = Analytics
