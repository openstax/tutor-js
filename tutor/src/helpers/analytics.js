import { partial, get } from 'lodash';

import Router from '../helpers/router';
import Courses from '../models/courses-map';

// generate custom event data for routes
const Events = {
  viewTaskStep({ courseId }) {
    // track all work done on a course
    return Analytics.sendEvent('Course', 'Work', { label: courseId });
  },

  viewStudentDashboard({ courseId }) {
    // compare activity between courses
    return Analytics.sendEvent('Student', 'Dashboard', { label: courseId });
  },
};

// a bit shorter helper methods
const isTeacher = courseId => get(Courses.get(courseId), 'currentRole.isTeacher', false);
const getRole = function(courseId) {
  if (Courses.get(courseId).is_teacher) { return 'teacher'; } else { return 'student'; }
};

const assignmentTypeTranslator = function(assignmentType, { courseId, id }) {
  const newOrEdit = id === 'new' ? 'create' : 'edit';
  return `/teacher/assignment/${newOrEdit}/${assignmentType}/${courseId}`;
};

function viewReferenceBook({ courseId, pageId }) {
  const url = `/reference-view/${courseId}`;
  if (!pageId) {
    return url;
  }
  const course = Courses.get(courseId);
  const page = course.referenceBook.pages.byId.get(pageId);
  return page ? `${url}/section/${page.chapter_section.key}` : url;
}

// Translators convert a url like '/foo/bar/123/baz/1' into a simplified one like just '/foo/bar'
const Translators = {
  dashboard({ courseId }) {
    if (isTeacher(courseId)) { return `/teacher/calendar/${courseId}`; } else { return `/student/dashboard/${courseId}`; }
  },
  practiceTopics({ courseId }) { return `/student/practice/${courseId}`; },
  viewPerformanceGuide({ courseId }) {
    return `/${getRole(courseId)}/performance-forecast/${courseId}`;
  },
  calendarByDate({ courseId }) { return `/teacher/calendar/${courseId}`; },
  viewGradebook({ courseId }) { return `/teacher/student-scores/${courseId}`; },
  courseSettings({ courseId }) { return `/teacher/roster/${courseId}`; },
  editAssignment: partial(assignmentTypeTranslator, 'assignment'),
  calendarViewPlanStats({ courseId }) { return `/teacher/metrics/quick/${courseId}`; },
  reviewTask({ courseId }) { return `/teacher/metrics/review/${courseId}`; },
  viewReferenceBookSection: viewReferenceBook,
  viewReferenceBookPage: viewReferenceBook,
  viewReferenceBook,

  // Task steps are viewed by both teacher and student with no difference in params
  viewTaskStep({ courseId }) {
    const course = Courses.get(courseId);
    const role = course ? course.primaryRole.type : 'unknown';
    return `/${role}/task-step/${courseId}`;
  },
};

const TAG_MANAGER_ID = 'GTM-W6N7PB';

let GA = undefined;
let trackerNames = undefined;

var Analytics = {
  // copied from BIT
  initTagManager(w, d, s, l, i) {
    // Disable ESLint rules since we're copying Google's script
    /* eslint-disable one-var, prefer-const, prefer-template */
    w[l] = w[l] || [];
    w[l].push({
      'gtm.start': new Date().getTime(),
      event: 'gtm.js',
    });
    let f = d.getElementsByTagName(s)[0], j = d.createElement(s), dl = l !== 'dataLayer' ? '&l=' + l : '';
    // Breaks in tests because there are no scripts
    if (f) {
      j.async = true;
      j.src = '//www.googletagmanager.com/gtm.js?id=' + i + dl;
      f.parentNode.insertBefore(j, f);
    }
  },
  
  // Set the Command Queue (ga) and cache the (possibly multiple) tracker names
  setGa(ga) {
    trackerNames = undefined;
    GA = ga;
    if (GA) {
      GA(function() {
        trackerNames = GA.getAll().map(function(tracker) { return tracker.get('name'); });
      });
      this.initTagManager(window, document, 'script', 'dataLayer', TAG_MANAGER_ID);
    }

    return GA;
  },

  // Calls all trackers with the command and arguments given here
  ga(command, ...params) {
    if (!GA || !trackerNames) { return; }
    trackerNames.forEach((trackerName) => {
      this.realGa(trackerName + '.' + command, ...params);
    });
  },

  // Exists largely for spec mocking
  realGa(...args) {
    GA(...args);
  },

  sendPageView(url) {
    this.ga('send', 'pageview', url);
  },

  recordCourseDimension(courseId) {
    const course = Courses.get(courseId);
    if (course) {
      this.ga('set', 'dimension1', course.appearance_code);
    }
  },

  onNavigation(path) {
    if (!GA) { return; }
    const route = Router.currentMatch(path);

    if (!route) {
      this.sendPageView(`/not-found${path}`);
      return;
    }
    const { courseId } = route.params;
    if (courseId) { this.recordCourseDimension(courseId); }
    const translatedPath = Translators[route.entry.name] ?
      Translators[route.entry.name]( route.params ) : route.pathname;

    this.ga('set', 'page', translatedPath);
    // if we're also going to send custom events then we set the page
    if (Events[route.entry.name]) {
      Events[route.entry.name]( route.params );
    }

    this.sendPageView(); // url's not needed since it was set before events
  },

  sendEvent(category, action, attrs = {}) {
    this.ga('send', 'event', category, action, attrs.label, attrs.value);
  },
};

export default Analytics;
