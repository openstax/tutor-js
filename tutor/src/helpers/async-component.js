import React from 'react';
import AsyncErrorBoundary from '../components/error-monitoring/async-load-error';
import LoadingAnimation from '../../../shared/src/components/loading-animation'

export function asyncComponent(loader, name = '') {

    const Component = React.lazy(loader);
    const loading = <LoadingAnimation message={`Loading ${name}â€¦`} />;

    const Loader = (props) => (
        <AsyncErrorBoundary>
            <React.Suspense fallback={loading}>
                <Component {...props} />
            </React.Suspense>
        </AsyncErrorBoundary>
    );

    return (props) => <Loader {...props} />;
}


export function loadAsync(resolve, name) {
    // return a function so the router will only evaluate it when it's needed
    // in the future we may insert role dependant logic here
    return () => asyncComponent(resolve, name);
}
