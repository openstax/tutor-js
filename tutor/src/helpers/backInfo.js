import { navHistory } from '../models'


const backInfo = () => {
    // Gets route to last path that was not the same as the current one
    // See TransitionStore for more detail.
    const historyInfo = navHistory.latest

    const text = historyInfo != null && !!historyInfo.name ? `Back to ${historyInfo.name}` : 'My Courses';
    const to =  historyInfo != null && !!historyInfo.path ? historyInfo.path : 'myCourses';

    return {
        text,
        to,
    };
};


export default backInfo;
