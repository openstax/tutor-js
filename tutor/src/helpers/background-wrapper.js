import { styled } from 'vendor';
import { navbars } from '../theme.js';

const BackgroundWrapper = styled.div`
  background: #fff;
  min-height: calc(100vh - ${navbars.top.height} - ${navbars.bottom.height});
  position: relative;
  overflow: hidden;
  padding: 0 2.4rem;
`;

const ContentWrapper = styled.div`
  max-width: 1200px;
  margin: 0 auto;
`;

export { BackgroundWrapper, ContentWrapper };
