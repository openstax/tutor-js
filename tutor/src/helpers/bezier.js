function EASE_IN_OUT(t) {
    if (t < .5) { return 4 * t * t * t; } else { return ((t - 1) * ((2 * t) - 2) * ((2 * t) - 2)) + 1; }
}

function CALC(start, end, elapsed, duration) {
    if (elapsed > duration) { return end; }
    return start + ((end - start) * EASE_IN_OUT(elapsed / duration));
}

export default function bezier({
    range, duration, onStep, onComplete,
}) {
    const startTime = Date.now();
    const requestAnimationFrame = window.requestAnimationFrame || window.setTimeout;

    const step = () => {
        const elapsed = Date.now() - startTime;
        onStep(CALC(range[0], range[1], elapsed, duration));
        if (elapsed < duration) {
            requestAnimationFrame(step);
        } else if (onComplete) {
            onComplete();
        }
    };

    step();
}
