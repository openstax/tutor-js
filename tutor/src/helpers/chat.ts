import { Raven } from '../models'

const DEPLOYMENT_ID   = '5724P000000ghju';
const ORGANIZATION_ID = '00DU0000000Kwch';
const ELEMENT_ID      = '5736f000000k9eQ';


const win = window as any

export const Chat = {
    initialized: false,

    get isEnabled() {
        return Boolean(win.liveagent);
    },

    get isOnline() {
        if (!this.isEnabled) { return false; }
        const chat = document.querySelector<HTMLDivElement>('.chat.enabled');
        return chat && chat.style.display !== 'none';
    },

    initialize() {
    // will emit debugging info to the console
    // liveagent.enableLogging();
    },

    setElementVisiblity(online: any, offline?: any) {
        if (!this.isEnabled) { return; }
        try {
            win.liveagent.showWhenOnline(ELEMENT_ID, online);
            if (offline) {
                win.liveagent.showWhenOffline(ELEMENT_ID, offline);
            }
            if (!this.initialized) {
                win.liveagent.init('https://d.la1-c2-ia5.salesforceliveagent.com/chat', DEPLOYMENT_ID, ORGANIZATION_ID);
                this.initialized = true;
            }
        } catch(err) {
            Raven.captureException(err);
        }
    },

    start() {
        win.liveagent.startChat(ELEMENT_ID);
    },

};
