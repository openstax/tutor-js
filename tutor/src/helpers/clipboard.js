import _ from 'underscore';

const SUPPORTED = typeof document.queryCommandSupported === 'function' ? document.queryCommandSupported('copy') : undefined;

export default {
    isSupported: _.constant(!!SUPPORTED),

    copy() {
        if (!SUPPORTED) { return; }

        try {
            document.execCommand('copy');
        } catch (e) {
            // eslint-disable-next-line no-console
            console.warn('clipboard copy failed', e);
        }
    },
};
