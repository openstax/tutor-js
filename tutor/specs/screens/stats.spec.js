import { R, ld, Factory, TimeMock } from '../helpers';
import Stats from '../../src/screens/stats';
import StatsStore from '../../src/models/stats';

// apex charts calls textNode.getBBox which jsdom doesn't support
jest.mock('react-apexcharts', () => (
  (props) => JSON.stringify(props)
));

describe('Stats Screen', () => {
  let props;

  TimeMock.setTo('2019-10-30T12:00:00.000Z');

  beforeEach(() => {
    const stats = new StatsStore();
    window.scrollTo = jest.fn();
    stats.onLoaded({
      data: ld.times(20, () => Factory.bot.create('Stat')),
    });
    stats.api.requestCounts.read = 1;
    props = {
      stats,
    };
  });

  it('renders and matches snapshot', () => {
    expect(<R><Stats {...props} /></R>).toMatchSnapshot();
  });

});
