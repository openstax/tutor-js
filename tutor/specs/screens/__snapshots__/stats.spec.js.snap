// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Screen renders and matches snapshot 1`] = `
<ForwardRef
  path="/"
  withRef={false}
>
  <Stats
    stats={
      Stats {
        "data": Array [
          Object {
            "ends_at": 1571832000000,
            "id": 1,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 87505,
              "active_populated_courses": 24085,
              "active_students": 39887,
              "exercise_steps": 15112,
              "highlights": 59317,
              "new_enrollments": 76276,
              "new_highlights": 29258,
              "notes": 5764,
              "reading_steps": 69170,
              "task_plans": 4302,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 2,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 2942,
              "active_populated_courses": 13089,
              "active_students": 10590,
              "exercise_steps": 66662,
              "highlights": 34345,
              "new_enrollments": 24024,
              "new_highlights": 55204,
              "notes": 36786,
              "reading_steps": 51312,
              "task_plans": 87350,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 3,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 19317,
              "active_populated_courses": 85445,
              "active_students": 55325,
              "exercise_steps": 84650,
              "highlights": 32198,
              "new_enrollments": 81792,
              "new_highlights": 29369,
              "notes": 88480,
              "reading_steps": 66156,
              "task_plans": 46023,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 4,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 80912,
              "active_populated_courses": 82911,
              "active_students": 17108,
              "exercise_steps": 35426,
              "highlights": 38483,
              "new_enrollments": 41799,
              "new_highlights": 86857,
              "notes": 55948,
              "reading_steps": 31678,
              "task_plans": 67792,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 5,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 52115,
              "active_populated_courses": 268,
              "active_students": 52153,
              "exercise_steps": 57855,
              "highlights": 33867,
              "new_enrollments": 30919,
              "new_highlights": 58063,
              "notes": 5893,
              "reading_steps": 55237,
              "task_plans": 47845,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 6,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 34924,
              "active_populated_courses": 52001,
              "active_students": 29248,
              "exercise_steps": 20763,
              "highlights": 98834,
              "new_enrollments": 44905,
              "new_highlights": 54109,
              "notes": 91988,
              "reading_steps": 90534,
              "task_plans": 65534,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 7,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 56348,
              "active_populated_courses": 80696,
              "active_students": 56435,
              "exercise_steps": 25754,
              "highlights": 90191,
              "new_enrollments": 51761,
              "new_highlights": 71519,
              "notes": 28232,
              "reading_steps": 98363,
              "task_plans": 29587,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 8,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 87311,
              "active_populated_courses": 86586,
              "active_students": 60069,
              "exercise_steps": 16106,
              "highlights": 39437,
              "new_enrollments": 98747,
              "new_highlights": 16788,
              "notes": 35204,
              "reading_steps": 73107,
              "task_plans": 63277,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 9,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 6546,
              "active_populated_courses": 45063,
              "active_students": 20454,
              "exercise_steps": 42834,
              "highlights": 98352,
              "new_enrollments": 88694,
              "new_highlights": 535,
              "notes": 34941,
              "reading_steps": 7936,
              "task_plans": 82620,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 10,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 2755,
              "active_populated_courses": 56900,
              "active_students": 92758,
              "exercise_steps": 29686,
              "highlights": 54776,
              "new_enrollments": 24182,
              "new_highlights": 81733,
              "notes": 91131,
              "reading_steps": 9332,
              "task_plans": 19940,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 11,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 47387,
              "active_populated_courses": 70869,
              "active_students": 4857,
              "exercise_steps": 74186,
              "highlights": 45741,
              "new_enrollments": 41679,
              "new_highlights": 50942,
              "notes": 5608,
              "reading_steps": 75352,
              "task_plans": 6756,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 12,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 80738,
              "active_populated_courses": 30646,
              "active_students": 28653,
              "exercise_steps": 78099,
              "highlights": 83924,
              "new_enrollments": 68618,
              "new_highlights": 81103,
              "notes": 25584,
              "reading_steps": 16593,
              "task_plans": 38151,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 13,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 19017,
              "active_populated_courses": 69631,
              "active_students": 88785,
              "exercise_steps": 66487,
              "highlights": 66526,
              "new_enrollments": 50236,
              "new_highlights": 48817,
              "notes": 99261,
              "reading_steps": 11139,
              "task_plans": 40518,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 14,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 4802,
              "active_populated_courses": 8490,
              "active_students": 56564,
              "exercise_steps": 76509,
              "highlights": 44032,
              "new_enrollments": 26591,
              "new_highlights": 60469,
              "notes": 40335,
              "reading_steps": 43821,
              "task_plans": 18109,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 15,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 47746,
              "active_populated_courses": 75071,
              "active_students": 92757,
              "exercise_steps": 33706,
              "highlights": 58267,
              "new_enrollments": 5661,
              "new_highlights": 17151,
              "notes": 88134,
              "reading_steps": 81484,
              "task_plans": 58404,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 16,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 44064,
              "active_populated_courses": 82150,
              "active_students": 85938,
              "exercise_steps": 7914,
              "highlights": 57406,
              "new_enrollments": 20587,
              "new_highlights": 68384,
              "notes": 15703,
              "reading_steps": 75164,
              "task_plans": 33861,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 17,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 7962,
              "active_populated_courses": 39937,
              "active_students": 13841,
              "exercise_steps": 8178,
              "highlights": 90987,
              "new_enrollments": 69856,
              "new_highlights": 48539,
              "notes": 61643,
              "reading_steps": 12863,
              "task_plans": 92447,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 18,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 43310,
              "active_populated_courses": 81164,
              "active_students": 20139,
              "exercise_steps": 12224,
              "highlights": 42430,
              "new_enrollments": 49574,
              "new_highlights": 49433,
              "notes": 8156,
              "reading_steps": 56221,
              "task_plans": 86910,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 19,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 84127,
              "active_populated_courses": 93193,
              "active_students": 55159,
              "exercise_steps": 742,
              "highlights": 46798,
              "new_enrollments": 46268,
              "new_highlights": 65552,
              "notes": 37072,
              "reading_steps": 80793,
              "task_plans": 92212,
            },
          },
          Object {
            "ends_at": 1571832000000,
            "id": 20,
            "starts_at": 1571227200000,
            "stats": Object {
              "active_courses": 28821,
              "active_populated_courses": 66838,
              "active_students": 37898,
              "exercise_steps": 71775,
              "highlights": 58217,
              "new_enrollments": 69073,
              "new_highlights": 26398,
              "notes": 56501,
              "reading_steps": 20609,
              "task_plans": 42391,
            },
          },
        ],
      }
    }
  />
</ForwardRef>
`;
